bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/magento/GraphqlDataServiceImpl.java

```diff
@@ -278,10 +278,9 @@ private CategoryTree getCategoriesImpl(Integer categoryId, String storeView) {
         // Search parameters
         CategoryArgumentsDefinition searchArgs = q -> q.id(categoryId);
 
-        // Main query
+        // Main query: we set a pageSize of 100'000 because there isn't any way to specify "unlimited"
         CategoryTreeQueryDefinition queryArgs = q -> q
-            .products(p -> p
-                .items(GraphqlQueries.CONFIGURABLE_PRODUCT_QUERY));
+            .products(o -> o.pageSize(100000), p -> p.items(GraphqlQueries.CONFIGURABLE_PRODUCT_QUERY));
```

Comment:
We should be very careful with these numbers. Among others, the query retrieves the product description in HTML, which potentially can be quite large. So if the backend indeed returns 100000 products, that's quite a lot of data and can impact performance and caching behaviour. 

As a quick fix, this is okay, but we should create a follow up issue to implement pagination here.